chore(artifacts/gitRepo): git repo artifact (#1569)
spinnaker/spinnaker.github.io: reference/artifacts-with-artifactsrewrite/types/git-repo/index.md

---
layout: single
title:  "Git Repo"
sidebar:
  nav: reference
---

{% include toc %}

Git Repo artifacts are references to Git repositories that are hosted by a Git hosting service. They are consumed
by stages that need multiple files to produce an output such as the Bake (Manifest) stage when using the Kustomize template
renderer. Unlike other artifact implementation, the Git Repo artifact will work with any Git hosting service as long as the
repository can be cloned using the Git CLI.

## Git Repo artifact in the UI

The pipeline UI exposes the following fields for the GitHub file artifact:

> <table>

Reviewer: Could we do this using a Markdown table, like the other table below?

---
layout: single
title:  "Git Repo"
sidebar:
  nav: reference
---

**The Git Repo artifact is available in version 1.17 or later.**
{: .notice--info}

{% include toc %}

Git Repo artifacts are references to Git repositories that are hosted by a Git hosting service. They are consumed
by stages that need multiple files to produce an output, such as the Bake (Manifest) stage when using the Kustomize template
renderer. Unlike other artifact implementations, the Git Repo artifact works with any Git hosting service as long as the
repository can be cloned using the Git CLI.

## Git Repo artifact in the UI

The pipeline UI exposes the following fields for the GitHub file artifact:

| Field | Explanation |
| ------|--------------|
|Account| A Git Repo artifact account. |

> |URL| HTTP or SSH URL of your Git repository. Artifact accounts configured to use a username and password should use HTTP while accounts configured to use an SSH private key should use the SSH URL format. |

Reviewer: ```suggestion
|URL| HTTP or SSH URL of your Git repository. Use HTTP for artifact accounts configured to use a username and password, and use the SSH URL format for accounts configured to use the SSH private key. |
```

|Branch| Name of the branch to check out. Default is `master`. |
|Sub Path (Optional) | Relative path of files within the repository to fetch. If set, only files this path will be available to stages consuming this artifact. |

### In a trigger

_Work to support Git Repo artifacts in triggers is currently underway._

### In a pipeline stage

When configuring the "Bake (Manifest)" stage, you can use a Git Repo artifact as an expected artifact.

{%
    include
    figure
    image_path="./bake-manifest-stage-git-repo.png"
    caption="Configure a Bake (Manifest) stage to use a Git Repo artifact."
%}

## Git Repo artifact in a pipeline definition

> | Field              | Explaination                                                                                                                                                                                            |

Reviewer: ```suggestion
| Field              | Explanation                                                                                                                                                                                            |
```

|--------------------|--------|
| `type`             | Always `git/repo`.                                                                                                                                                                                       |

> | `reference`        |  HTTPS or SSH URL of your Git repository. Artifact accounts configured to use a username and password should use HTTPS while accounts configured to use an SSH private key should use the SSH URL format. |

Reviewer: ```suggestion
| `reference`        |  HTTPS or SSH URL of your Git repository. Use HTTPS for artifact accounts configured to use a username and password, and use the SSH URL format for accounts configured to use an SSH private key. |
```
